homeassistant/components/rainbird/binary_sensor.py

```diff
@@ -33,22 +43,24 @@
 )
 
 
-async def async_setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    config_entry: ConfigEntry,
+    async_add_devices: AddEntitiesCallback,
 ) -> None:
-    """Set up a Rain Bird sensor."""
-    if discovery_info is None:
-        return
-
-    async_add_entities(
-        [
-            RainBirdSensor(discovery_info[description.key], description)
-            for description in BINARY_SENSOR_TYPES
+    """Set up entry for a Rain Bird binary_sensor."""
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    await asyncio.gather(
+        *[
+            data[description.key].async_config_entry_first_refresh()
```

Comment:
This won't work here as we're not inside the config entry setup anymore. Why don't we do this in the config entry setup in the init module?

Reply:
I've changed this to use a single coordinator (more in other comment)